Mailbox/DataLoader.cs

```diff
@@ -1,21 +1,81 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.IO;
+using Newtonsoft.Json;
 
 namespace Mailbox
 {
-    public class DataLoader
+    public class DataLoader : IDisposable
     {
+        private readonly Stream Source;
+        private bool StreamDisposed = false;
+
         public DataLoader(Stream source)
         {
+            if(source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+            Source = source;
         }
 
-        public List<Mailbox> Load()
+        public List<Mailbox>? Load()
         {
+            List<Mailbox> mailboxes = new List<Mailbox>();
+            Source.Position = 0;
+
+            try
+            {
+                using (StreamReader reader = new StreamReader(Source))
```

Comment:
This will cause the stream to get closed.
```suggestion
                using (StreamReader reader = new StreamReader(Source, leaveOpen:true))
```